TUTORIAL.md

```diff
@@ -0,0 +1,392 @@
+
+The first failing test is in our Course class, `undefined method create_table for Course:Class.` 
+
+If you take a look at `lib/student.rb`, you will see there is a `create_table` method already written. The process for creating a new table will be similar for all of the classes you build, minus a few key details. For exampe, the `SQL` statement for our student class reads `CREATE TABLE IF NOT EXISTS students`. To get this to work for our course class, we will have to change it to `CREATE TABLE IF NOT EXISTS courses`. Don't forget to use commas between your `id`, `name` and `department_id` attributes, otherwise it will be invalid code and your tests will not pass. Compare your Student and Course classes, see the difference? The last thing we need to do is add our reader and writer methods, or `attr_accessors` for `:id`, `:name` and `:department_id`. Our 
+
+***Note: `create_table` and `drop_table` are tested together in the specs, so you will have to build both of them as well as add your `attr_accessors` before your tests pass.***
+
+##lib/course.rb
+
+```ruby
+class Course
+attr_accessor :id, :name, :department_id
+
+  def self.create_table
+    sql = <<-SQL
+    CREATE TABLE IF NOT EXISTS courses (
+      id INTEGER PRIMARY KEY,
+      name TEXT,
+      department_id INTEGER 
+    )
+    SQL
+     
+    DB[:conn].execute(sql)
+  end
+  
+end
+
+```
+We now have a way to create a table, but what if we want to drop a table? Yep, we need a `drop_table` method to complete the circuit. Take a look at your `lib/student.rb` and you will see a `drop_table` method for students, use that as your guide. This method is pretty clear, if a table exists, drop it when the method is executed. Just like our `create_table` method, we need to prepend `self`, which accesses our Class scope. Since the courses table is responsible for storing all the courses in the Course class, it makes sense for this to be a class method.
+
+```ruby
+def self.drop_table
+  sql = "DROP TABLE IF EXISTS courses"
+  DB[:conn].execute(sql)
+end
+```
+Run `rspec` and your will see the next error, `undefined method insert`. Again, lets take a look at our `student.rb` file and see if we have reusable code. 
+
+It looks like we have an `insert` method built for us so lets modify that and use it. Let's also create a helper method called `attribute_values` to keep everything organized. Instead of typing out all of the values in our insert method, we can call the `attribute_values` method instead.
```

Comment:
Wait how did you know that the next thing to work on was the insert method? What's the goal behind modifiying it?